src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -831,6 +834,20 @@ private void NotifyCertificateSelection(MergedOptions mergedOptions, IConfidenti
             }
         }
 
+        /// <summary>
+        /// Checks assertion and sub_assertion passed from merging extra query parameters to ensure they do not contain unsupported characters.
+        /// </summary>
+        /// <param name="assertion">The assertion.</param>
+        /// <param name="subAssertion">The sub_assertion.</param>
+        private static void CheckAssertionsForInjectionAttempt(string assertion, string subAssertion)
+        {
+            if (!assertion.IsNullOrEmpty() || !subAssertion.IsNullOrEmpty())
+            {
+                if (assertion.Contains('&', StringComparison.InvariantCultureIgnoreCase)) throw new ArgumentException(IDWebErrorMessage.InvalidAssertion, nameof(assertion));
```

Comment:
Won't you still have a null reference exception if assertion is null, but not subAssertion?
Shouldn't it be?

```csharp
 if (!assertion.IsNullOrEmpty()  && assertion.Contains('&', StringComparison.InvariantCultureIgnoreCase)) throw new ArgumentException(IDWebErrorMessage.InvalidAssertion, nameof(assertion));
if (!subAssertion.IsNullOrEmpty() && (subAssertion.Contains('&', StringComparison.InvariantCultureIgnoreCase)) throw new ArgumentException(IDWebErrorMessage.InvalidSubAssertion, nameof(subAssertion));
```

Reply:
I believe this is taken care of in line 780, 781 and 784 the only reason I even did the check was due to warnings from Nullable checks.

Reply:
What @jmprieur suggests will work. By checking for null before calling the `IsNullOrEmpty()` method, you can ensure that the method is only called on non-null strings

Reply:
Thanks @jmprieur & @jennyf19